bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/ItemToStorePointCreator.java

```diff
@@ -0,0 +1,140 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.Metadata;
+import org.openhab.core.items.MetadataKey;
+import org.openhab.core.items.MetadataRegistry;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.openhab.persistence.influxdb.InfluxDBPersistenceService;
+
+/**
+ * Logic to create an InfluxDB {@link InfluxPoint} from an openHAB {@link Item}
+ *
+ * @author Joan Pujol Espinar - Initial contribution
+ */
+@NonNullByDefault
+public class ItemToStorePointCreator {
+    private final InfluxDBConfiguration configuration;
+    @Nullable
+    private final MetadataRegistry metadataRegistry;
+
+    public ItemToStorePointCreator(InfluxDBConfiguration configuration, @Nullable MetadataRegistry metadataRegistry) {
+        this.configuration = configuration;
+        this.metadataRegistry = metadataRegistry;
+    }
+
+    public @Nullable InfluxPoint convert(Item item, @Nullable String storeAlias) {
+        if (item.getState() instanceof UnDefType) {
+            return null;
+        }
+
+        String measurementName = calculateMeasurementName(item, storeAlias);
+        String itemName = item.getName();
+        State state = getItemState(item);
+
+        Object value = InfluxDBStateConvertUtils.stateToObject(state);
+
+        InfluxPoint.Builder point = InfluxPoint.newBuilder(measurementName).withTime(Instant.now()).withValue(value)
+                .withTag(TAG_ITEM_NAME, itemName);
+
+        addPointTags(item, point);
+
+        return point.build();
+    }
+
+    @SuppressWarnings("null")
+    private String calculateMeasurementName(Item item, @Nullable String storeAlias) {
+        String name;
+        if (StringUtils.isNotBlank(storeAlias)) {
+            name = storeAlias;
+        } else {
+            name = item.getName();
+        }
+
+        if (configuration.isReplaceUnderscore()) {
+            name = name.replace('_', '.');
+        }
+
+        return name;
+    }
+
+    private State getItemState(Item item) {
+        final State state;
+        final Class<? extends State> desiredConversion = calculateDesiredTypeConversionToStore(item);
+        if (desiredConversion != null) {
+            State convertedState = item.getStateAs(desiredConversion);
+            if (convertedState != null) {
+                state = convertedState;
+            } else {
+                state = item.getState();
+            }
+        } else {
+            state = item.getState();
+        }
+        return state;
+    }
+
+    private @Nullable Class<? extends State> calculateDesiredTypeConversionToStore(Item item) {
+        Class<? extends State> conversion = null;
+        if (item.getAcceptedCommandTypes().size() > 1) {
+            if (item.getAcceptedCommandTypes().get(0).isAssignableFrom(State.class)) {
+                conversion = item.getAcceptedCommandTypes().get(0).asSubclass(State.class);
+            }
```

Comment:
Shouldn't you be looping through the list of accepted command types? Or am I missing something?